tools/compiletest/src/main.rs

```diff
@@ -385,6 +386,46 @@ fn collect_expected_tests_from_dir(
     Ok(())
 }
 
+/// Collect `exec` tests from a directory.
+///
+/// Note that this method isn't recursive like
+/// `collect_expected_tests_from_dir`, as it allows us to ensure that newly
```

Comment:
It is nice to be able to check that newly tests are running, but the inability to add tests under sub-directories could be a bit restrictive.

Reply:
Yes, but I wouldn't like to allow tests like these ones (which are quite complex) to be missed. Of course, this can be changed if it becomes too restrictive.